### PR TITLE
Fix headers and schema definition

### PR DESCRIPTION
**Pull request by wangbwn**
Fix headers and schema definition.

### CODE REVIEW

**Review by Dreamgoing — APPROVED** [review]

**Review by ruowan — APPROVED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 1]
```diff
@@ -8045,8 +8048,11 @@
         },
         "headers": {
           "type": "object",
-          "x-ms-format": "dfe-key-value-pairs",
-          "description": "Represents the headers that will be sent to the request. For example, to set the language and type on a request: \"headers\" : { \"Accept-Language\": \"en-us\", \"Content-Type\": \"application/json\" }. Type: dictionary (or Expression with resultType dictionary)."
+          "description": "Represents the headers that will be sent to the request. For example, to set the language and type on a request: \"headers\" : { \"Accept-Language\": \"en-us\", \"Content-Type\": \"application/json\" }. Type: string (or Expression with resultType string).",
+          "additionalProperties": {
+            "type": "string",
+            "x-ms-format": "dfe-string"
```
Comment (Frey-Wang) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8054]: If you use x-ms-format, then its type must be object.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 2]
```diff
@@ -8111,8 +8117,11 @@
...
```
Comment (Frey-Wang) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8123]: Same here

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json (hunk 1 above)
Comment (ruowan) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8054]: According to our test, if we define the type as object. For additionalProperties the generated model is `public IDictionary<string, BinaryData> Headers { get; }` and it's not our expected.

**Comment by ruowan** [comment]
This is just a minor changes. This PR is for fixing wrong swagger definition. There is no breaking change for customer.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 3]
```diff
@@ -8045,8 +8048,11 @@
         },
         "headers": {
           "type": "object",
-          "x-ms-format": "dfe-key-value-pairs",
-          "description": "Represents the headers that will be sent to the request. For example, to set the language and type on a request: \"headers\" : { \"Accept-Language\": \"en-us\", \"Content-Type\": \"application/json\" }. Type: dictionary (or Expression with resultType dictionary)."
+          "description": "Represents the headers that will be sent to the request. For example, to set the language and type on a request: \"headers\" : { \"Accept-Language\": \"en-us\", \"Content-Type\": \"application/json\" }. Type: string (or Expression with resultType string).",
+          "additionalProperties": {
```
Comment (mentat9) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8052]: >additionalProperties [](http://example.com/codeflow?start=11&length=20)

Use of additionalProperties is not allowed for properties owned by the service. The only time its ok to use it is when the properties are pass thru (user defined) and not subject to any validations. Please schematize this object, make it an array or provide an explanation for why you need to use this pattern.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 4]
```diff
@@ -8111,8 +8117,11 @@
...
```
Comment (mentat9) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8121]: >additionalProperties [](http://example.com/codeflow?start=11&length=20)

Same here.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 5]
```diff
@@ -7007,8 +7007,11 @@
...
```
Comment (mentat9) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:7011]: >additionalProperties [](http://example.com/codeflow?start=11&length=20)

Use of additionalProperties is not allowed for properties owned by the service. The only time its ok to use it is when the properties are pass thru (user defined) and not subject to any validations. Please schematize this object, make it an array or provide an explanation for why you need to use this pattern.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json (hunk 3 above)
Comment (ruowan) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8052]: The additionalProperties we've included serve as HTTP headers specified by the customer. This schema does not follow a specific pattern.

Moreover, the ADF SDK has some unique x-ms-format, identified by a 'dfe-' prefix, such as 'dfe-key-value-pairs'. This update in the schema also addresses an issue in the .NET Track2 SDK, where customers were unable to create web activities with headers (refer to [Azure SDK for .NET Issue #39187](https://github.com/Azure/azure-sdk-for-net/issues/39187) for more details). We've conducted local tests, and the solution appears to be functioning effectively.

Comment (mentat9) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8052]: Are the HTTP headers just passed straight through without any examination or validation applied?

Comment (ruowan) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:8052]: Yes.  The HTTP headers just passed through. No validation.

**Comment by ruowan** [comment]
@mentat9 our customer is waiting for new fixed SDK release. Could you pls approve and merge this PR?  our customer is waiting for new fixed SDK release.

**Comment by ruowan** [comment]
@mentat9 breaking change has been approved. Could you pls approve this PR?

**Pull request merged**